### PR TITLE
[MeshMovingApp] split core and py-interface

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
see title
Lets see if I added `KRATOS_API` in the correct places

### CODE REVIEW

File: applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.h [hunk 1]
```diff
@@ -28,20 +28,20 @@ namespace MoveMeshUtilities {
...
 
-void CheckJacobianDimension(GeometryType::JacobiansType &rInvJ0,
+void KRATOS_API(MESH_MOVING_APPLICATION) CheckJacobianDimension(GeometryType::JacobiansType &rInvJ0,
                             VectorType &rDetJ0, const GeometryType &rGeometry);
 
-void MoveMesh(ModelPart::NodesContainerType &rNodes);
+void KRATOS_API(MESH_MOVING_APPLICATION) MoveMesh(ModelPart::NodesContainerType &rNodes);
 
-ModelPart* GenerateMeshPart(ModelPart &rModelPart,
+ModelPart* KRATOS_API(MESH_MOVING_APPLICATION) GenerateMeshPart(ModelPart &rModelPart,
```
Comment (philbucher) [inline applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.h]: @roigcarlo it seems to not like this function, it works neither with nor without `KRATOS_API`
any idea?

Comment (philbucher) [reply applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.h]: maybe it just doesn't work with such types, I could refactor the function...

Comment (roigcarlo) [reply applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.h]: The * is the problem, it has to be either:
```C++
KRATOS_API(MESH_MOVING_APPLICATION) ModelPart* GenerateMeshPart
```
```C++
// Please don't use this one.
ModelPart KRATOS_API(MESH_MOVING_APPLICATION) * GenerateMeshPart
```
or 
```C++
typedf ModelPart* ModelPartPtr;
ModelPartPtr KRATOS_API(MESH_MOVING_APPLICATION) GenerateMeshPart
```

Comment (philbucher) [reply applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.h]: cool thx I try

Comment (philbucher) [reply applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.h]: worked, thx

**Review by roigcarlo — APPROVED** [review]
:+1:

**Pull request merged**